ee/hogai/eval/schema.py

```diff
@@ -0,0 +1,143 @@
+import json
+from abc import ABC, abstractmethod
+from collections.abc import Generator, Sequence
+from typing import Generic, Self, TypeVar
+
+from django.db.models import Model
+from pydantic import BaseModel
+from pydantic_avro import AvroBase
+
+from posthog.models import (
+    DataWarehouseTable,
+    GroupTypeMapping,
+    PropertyDefinition,
+    Team,
+)
+
+T = TypeVar("T", bound=Model)
+
+
+class BaseSnapshot(AvroBase, ABC, Generic[T]):
+    @classmethod
+    @abstractmethod
+    def serialize_for_project(cls, project_id: int) -> Generator[Self, None, None]:
+        raise NotImplementedError
+
+    @classmethod
+    @abstractmethod
+    def deserialize_for_project(
+        cls, project_id: int, models: Sequence[Self], *, team_id: int
+    ) -> Generator[T, None, None]:
+        raise NotImplementedError
+
+
+# posthog/models/team.py
+class TeamSnapshot(BaseSnapshot[Team]):
+    name: str
+    test_account_filters: str
+
+    @classmethod
+    def serialize_for_project(cls, project_id: int):
+        team = Team.objects.get(pk=project_id)
```

Comment:
Same-ish question here - can we always guarantee that project_id exists? I assume the serialization happens right away, so there's not enough time for anything to happen in between.

However, from my understanding, they are refreshed twice monthly, so the question stands :)